src/packaging/markers.py

```diff
@@ -309,6 +309,12 @@ def evaluate(self, environment: dict[str, str] | None = None) -> bool:
         """
         current_environment = cast("dict[str, str]", default_environment())
         current_environment["extra"] = ""
+        # Work around platform.python_version() returning something that is not PEP 440
+        # compliant for non-tagged Python builds. We preserve default_environment()'s
+        # behavior of returning platform.python_version() verbatim, and leave it to the
+        # caller to provide a syntactically valid version if they want to override it.
```

Comment:
What was the motivation for preserving the non-compliant version number and expecting the caller to do this same logic? That leads to other callers like pipx encountering the same bug. Is the expectation that all downstream users should have to provide their own fix?